tardis/transport/montecarlo/formal_integral_cuda.py

```diff
@@ -236,52 +442,75 @@ def formal_integral(
         """
         Simple wrapper for the CUDA implementation of the formal integral
         """
-        L = np.zeros(inu_size, dtype=np.float64)  # array(float64, 1d, C)
         # global read-only values
         size_line, size_shell = tau_sobolev.shape  # int64, int64
         size_tau = size_line * size_shell  # int64
-
+        
         pp = np.zeros(N, dtype=np.float64)  # array(float64, 1d, C)
         exp_tau = np.zeros(size_tau, dtype=np.float64)  # array(float64, 1d, C)
         exp_tau = np.exp(-tau_sobolev.T.ravel())  # array(float64, 1d, C)
         pp[::] = calculate_p_values(
             self.geometry.r_outer[size_shell - 1], N
         )  # array(float64, 1d, C)
-
-        I_nu = np.zeros(
-            (inu_size, N), dtype=np.float64
-        )  # array(float64, 1d, C)
         z = np.zeros(
             (inu_size, 2 * size_shell), dtype=np.float64
         )  # array(float64, 1d, C)
         shell_id = np.zeros(
             (inu_size, 2 * size_shell), dtype=np.int64
         )  # array(int64, 1d, C)
-        THREADS_PER_BLOCK = 32
-        blocks_per_grid = (inu_size // THREADS_PER_BLOCK) + 1
 
-        cuda_formal_integral[blocks_per_grid, THREADS_PER_BLOCK](
-            self.geometry.r_inner,
-            self.geometry.r_outer,
+        # These get separate names since they'll be copied back
+        d_L = cuda.device_array((inu_size,), dtype=np.float64)
```

Comment:
Separate but not very descriptive...

Reply:
Generally, it is standard to prepend device objects with a 'd'

Reply:
I'll make a note